core/config/toml/types.go

```diff
@@ -1478,16 +1480,64 @@ func (drl *DispatcherRateLimit) setFrom(f *DispatcherRateLimit) {
 	}
 }
 
+type GatewayConnectorConfig struct {
+	ChainIDForNodeKey         *string
+	NodeAddress               *string
+	DonID                     *string
+	Gateways                  []ConnectorGatewayConfig
+	WsHandshakeTimeoutMillis  *uint32
+	AuthMinChallengeLen       *int
+	AuthTimestampToleranceSec *uint32
+}
+
+func (r *GatewayConnectorConfig) setFrom(f *GatewayConnectorConfig) {
+	if f.ChainIDForNodeKey != nil {
+		r.ChainIDForNodeKey = f.ChainIDForNodeKey
+	}
+
+	if f.NodeAddress != nil {
+		r.NodeAddress = f.NodeAddress
+	}
+
+	if f.DonID != nil {
+		r.DonID = f.DonID
+	}
+
+	// TODO: verify this copy by reference is ok, or does array need to be copied by value
```

Comment:
I commented this on the previous PR: it's fine to use the reference here, but the interface implementation (the getter) should do a deep copy before returning it. This will prevent bugs with a caller modifying the underlying array.